scripts/genpinctrl/config-f1.yaml

```diff
@@ -0,0 +1,156 @@
+#     be set according to the following rules:
+#       * Pin operates in analog mode: analog
+#       * Pin operates as an alternate input: input
+#       * Pin operates as an alternate output: alternate
+#       * Pin operates as an alternate bidirectional: alternate
+#
+#   - bias (optional): Bias setting (disable, pull-up, pull-down). This setting
+#     only applies to input mode. If not set pinctrl file will not contain the
+#     bias entry, equivalent to disable setting.
+#
+#   - drive (optional): Drive setting (push-pull, open-drain). If not set
+#     pinctrl file will not contain the drive entry, equivalent to push-pull
+#     setting
+#
+#   - variant (optional): Define another variant of a pin configuration. This
+#     setting can be useful to define multiple configurations for a certain
+#     alternate function.
+#
+
+---
+- name: ADC_IN
+  match: "^ADC\\d+_IN\\d+$"
+  mode: analog
+
+- name: CAN_RX
+  match: "^CAN\\d+_RX$"
+  mode: input
+
+- name: CAN_TX
+  match: "^CAN\\d+_TX$"
+  mode: alternate
+
+- name: DAC_OUT
+  match: "^DAC_OUT\\d+$"
+  mode: analog
+
+- name: I2C_SCL
+  match: "^I2C\\d+_SCL$"
+  drive: open-drain
+  mode: alternate
+
+- name: I2C_SDA
+  match: "^I2C\\d+_SDA$"
+  drive: open-drain
+  mode: alternate
+
+- name: I2S_CK
+  match: "^I2S\\d+_CK$"
+  mode: alternate
+
+- name: I2S_WS
+  match: "^I2S\\d+_WS$"
+  mode: alternate
+
+- name: I2S_SD
+  match: "^I2S\\d+_SD$"
+  mode: alternate
+
+- name: SPI_MASTER_MISO
+  match: "^SPI\\d+_MISO$"
+  bias: pull-down
+  mode: input
+  variant: master
+
+- name: SPI_MASTER_MOSI
+  match: "^SPI\\d+_MOSI$"
+  mode: alternate
+  variant: master
+
+- name: SPI_MASTER_SCK
+  match: "^SPI\\d+_SCK$"
+  mode: alternate
+  variant: master
+
+- name: SPI_MASTER_NSS
+  match: "^SPI\\d+_NSS$"
+  mode: alternate
+  variant: master
+
+- name: SPI_SLAVE_MISO
+  match: "^SPI\\d+_MISO$"
+  mode: alternate
+  variant: slave
+
+- name: SPI_SLAVE_MOSI
+  match: "^SPI\\d+_MOSI$"
+  bias: pull-down
+  mode: input
+  variant: slave
+
+- name: SPI_SLAVE_SCK
+  match: "^SPI\\d+_SCK$"
+  bias: pull-down
+  mode: input
+  variant: slave
+
+- name: SPI_SLAVE_NSS
+  match: "^SPI\\d+_NSS$"
+  bias: pull-up
+  mode: input
+  variant: slave
+
+- name: TIM_CH / TIM_CHN
+  match: "^TIM\\d+_CH\\d+N?$"
+  mode: alternate
+
+- name: TIM_BKIN
+  match: "^TIM\\d+_BKIN\\d?$"
+  bias: pull-up
+  mode: input
+
+- name: TIM_ETR
+  match: "^TIM\\d+_ETR$"
+  mode: input
+
+- name: UART_CTS / USART_CTS
+  match: "^US?ART\\d+_CTS$"
+  drive: open-drain
+  bias: pull-up
+  mode: input
+
+- name: UART_RTS / USART_RTS
+  match: "^US?ART\\d+_RTS$"
+  drive: open-drain
+  mode: alternate
+
+- name: UART_TX / USART_TX
+  match: "^US?ART\\d+_TX$"
+  mode: alternate
+
+- name: UART_RX / USART_RX
+  match: "^US?ART\\d+_RX$"
+  mode: input
+
+- name: USB_OTG_FS_DM
+  match: "^USB_OTG_FS_DM$"
+  mode: alternate
+
+- name: USB_OTG_FS_DP
+  match: "^USB_OTG_FS_DP$"
+  mode: alternate
```

Comment:
Here refman says: "Controlled automatically by the USB power-down".
Current code: `#define STM32_PIN_USB			(STM32_MODE_INPUT | STM32_CNF_IN_PUPD)`

Not sure what to do here.